### PR TITLE
Use enum for status states and fix setting of status

### PR DESCRIPTION
**Pull request by lbarcziova**
- implements using of `CommitStatus`enum in statuses
- fixes https://sentry.io/share/issue/15c35a842b2d47aebe971dfa31a6b373/

### CODE REVIEW

File: packit_service/worker/handlers/github_handlers.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
 from packit.local_project import LocalProject
 
 from packit_service.config import ServiceConfig, GithubPackageConfigGetter
+from packit_service.constants import GITHUB_FAILURE_STATUS
```
Comment (lachmanfrantisek) [inline packit_service/worker/handlers/github_handlers.py]: Please, use the `CommitStatus` enum instead:

https://github.com/packit-service/ogr/blob/63774270a240f9bd7064d33259417e7564a1e1a4/ogr/abstract.py#L536-L542

**Comment by TomasTomecek** [comment]
oh my, so there's more?! thanks for fixing this, I'm dumb

File: packit_service/worker/handlers/github_handlers.py (hunk 1 above)
Comment (lbarcziova) [reply packit_service/worker/handlers/github_handlers.py]: I will refactor it

Comment (lachmanfrantisek) [reply packit_service/worker/handlers/github_handlers.py]: Thanks!

File: packit_service/worker/build/build_helper.py [hunk 2]
```diff
@@ -240,7 +240,9 @@ def _report(
             description=description, state=state, url=url, check_names=check_names,
         )
 
-    def report_status_to_all(self, description: str, state: str, url: str = "") -> None:
+    def report_status_to_all(
+        self, description: str, state: Union[CommitStatus, str], url: str = ""
```
Comment (lachmanfrantisek) [inline packit_service/worker/build/build_helper.py]: It would be nice to not allow `str` and let the `mypy` find the wrong calls...

Comment (lbarcziova) [reply packit_service/worker/build/build_helper.py]: changed

**Pull request merged**